simple_graph/test_wgraph.py

```diff
@@ -0,0 +1,172 @@
+import pytest
+import weighted_graph as wgraph
+from collections import OrderedDict
+
+
+def test___init__():
+    """Test instantiates simple graph with dictionary."""
+    assert wgraph.Wgraph().dict == {}
+
+
+def test_nodes_empty(empty_graph):
+    """Test return list of all nodes in an empty graph"""
+    assert empty_graph.nodes() == []
+
+
+def test_nodes_populated(populated_graph):
+    """Test return list of all nodes in a populated graph"""
+    l = ['a', 'b', 'c', 'd', 'e']
+    sortedlist = populated_graph.nodes()
+    sortedlist.sort()
+    assert sortedlist == l
+
+
+def test_edges(populated_graph):
+    """Test return a list of all edges in the graph"""
+    edges = [('a', 'b'), ('b', 'c'), ('c', 'b'),
+             ('c', 'a'), ('b', 'd'), ('d', 'c')]
+    for edge in edges:
+        assert edge in populated_graph.edges()
+
+
+def test_add_node_empty(empty_graph):
+    """Test adds a new node 'n' to the empty graph"""
+    empty_graph.add_node('a')
+    assert empty_graph.has_node('a')
+
+
+def test_add_node_populated(populated_graph):
+    """Test adds a new node 'n' to the populated graph"""
+    populated_graph.add_node('g')
+    assert populated_graph.has_node('g')
+
+
+def test_has_node(empty_graph, populated_graph):
+    """Test True if node 'n' is contained in the graph, False if not."""
+    assert empty_graph.has_node('a') is False
+    l = ['a', 'b', 'c', 'd', 'e']
+    for node in l:
+        assert populated_graph.has_node(node) is True
+    assert populated_graph.has_node('t') is False
```

Comment:
This test should be two separate tests, one for the empty graph and one for the populated one.